realm/src/androidTest/java/io/realm/RealmTest.java

```diff
@@ -1056,4 +1056,27 @@ public void testWriteEncryptedCopy() throws Exception {
         }
     }
 
+    public void testWrongKeyShouldFail() {
+        final String WRONG_KEY_REALM = "wrong-key-realm.realm";
+
+        // Wrong key size
+        try {
+            Realm.getInstance(getContext(), WRONG_KEY_REALM, new byte[63]);
+            fail();
+        } catch (IllegalArgumentException ignored) {
+        } catch (Exception ignored) {
```

Comment:
you don't need to catch Exception - it would fail if we don't catch it. (although it would not test the other cases - but I'm fine with that)